frontend/covid-dashboard-frontend/src/components/HomePage.js

```diff
@@ -37,155 +35,185 @@ const HomePage = ({cities, states, countries, loading, cityDetail, stateDetail,
   const [testingRateOptions, setTestingRateOptions] = useState({});
   const [isReady, setIsReady] = useState(false);
 
+  const chartContainerHeight = '375px';
 
+  const CITY_LOCATION = 'city';
+  const STATE_LOCATION = 'state';
+  const COUNTRY_LOCATION = 'country';
 
   useEffect(() => {
     refreshCharts();
-  }, []);
+  }, [cityDetail, stateDetail, countryDetail, loading]);
 
-  const formatLocations = (items, key) => {
-    return items.map(item => {
-      return { ...item, value: item.location_id, label: item[key] };
-    });
+  const makeLocationLabel = (item, loc) => {
+    switch(loc) {
+      case CITY_LOCATION: return `${item.city}, ${item.state}, ${item.country}`;
+      case STATE_LOCATION: return `${item.state}, ${item.country}`;
+      case COUNTRY_LOCATION: return item.country;
+    }
   }
 
-  const makeDummyData = () => {
-    return {
-      xAxis: {
-        type: 'category',
-        data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
-      },
-      yAxis: {
-          type: 'value'
-      },
-      series: [{
-          data: [820, 932, 901, 934, 1290, 1330, 1320],
-          type: 'line'
-      }]
-    };
-  };
-
-  const commonToolBox = () => {
-    return {
-      feature: {
-        saveAsImage: { title: 'Save As Image' },
-        dataView: { readOnly: false, title: 'Data', lang: ['', 'Close', 'Refresh'] }
-      }
-    }
+  const formatLocations = (items, loc) => {
+    return items.map(item => {
+      return { ...item, value: item.location_id, label: makeLocationLabel(item, loc) };
+    });
```

Comment:
The /cities/ API now return city (aka county), state, country ... and /state/ API returns state, country so this updates the label for the searchable drop down to show the complete location in these cases